internal/buildscripts/packaging/msi/SplunkCustomActions/src/CustomActions.cs

```diff
@@ -44,6 +44,49 @@ public static ActionResult CheckLaunchConditions(Session session)
         return ActionResult.Success;
     }
 
+    /// <summary>
+    /// Custom action to add optional configuration options to the environment at the service scope.
+    /// </summary>
+    /// <remarks>
+    /// It is possible to add these as machine-wide environment variables, but, for consistency,
+    /// this custom action adds them to the service environment.
+    /// </remarks>
+    /// <param name="session">Carries information about the current installation session.</param>
+    /// <returns>Always success, in case of error the code will throw an exception and fails the install.</returns>
+    [CustomAction]
+    public static ActionResult AddOptionalConfigurations(Session session)
+    {
+        var optionalConfigurationKeys = new string[]
+        {
+            "SPLUNK_COLLECTD_DIR",
```

Comment:
Is this supposed to be a comprehensive list of env variables which could be looked up by collector?